graylog2-server/src/main/java/org/graylog/events/processor/aggregation/AggregationEventProcessor.java

```diff
@@ -168,6 +179,12 @@ private void filterSearch(EventFactory eventFactory, AggregationEventProcessorPa
                 final Message msg = resultMessage.getMessage();
                 final Event event = eventFactory.createEvent(eventDefinition, msg.getTimestamp(), eventDefinition.title());
                 event.setOriginContext(EventOriginContext.elasticsearchMessage(resultMessage.getIndex(), msg.getId()));
+
+                // We don't want source streams in the event which are unrelated to the event definition
+                msg.getStreamIds().stream()
+                    .filter(stream -> config.streams().contains(stream))
```

Comment:
Hmm, but shouldn't that use `streams` in case it was overloaded with parameters?

Reply:
Good catch, yes! :+1: